crates/edr_provider/src/lib.rs

```diff
@@ -514,3 +518,70 @@ fn to_json_with_traces<T: serde::Serialize, LoggerErrorT: Debug>(
         traces: value.1,
     })
 }
+
+// Rust port of https://github.com/NomicFoundation/hardhat/blob/024d72b09c6edefb00c012e9514a0948c255d0ab/v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/utils/convert-to-edr.ts#L176
+/// This normalization is done because this is the format Hardhat expects
+fn normalise_rpc_debug_trace<LoggerErrorT: Debug>(
+    value: (edr_evm::DebugTraceResult, Vec<Trace>),
+) -> Result<(RpcDebugTraceResult, Vec<Trace>), ProviderError<LoggerErrorT>> {
+    let trace = value.0;
+
+    let mut struct_logs = Vec::new();
+
+    for log in trace.logs {
+        let rpc_log = RpcDebugTraceLogItem {
+            pc: log.pc,
+            op: log.op_name.clone(),
+            op_name: log.op_name,
+            gas: u64::from_str_radix(log.gas.trim_start_matches("0x"), 16).unwrap_or(0),
+            gas_cost: u64::from_str_radix(log.gas_cost.trim_start_matches("0x"), 16).unwrap_or(0),
+            stack: log.stack.map(|values| {
+                values
+                    .into_iter()
+                    // Removing this trim temporarily as the Hardhat test assumes 0x is there
+                    // .map(|value| value.trim_start_matches("0x").to_string())
```

Comment:
If Hardhat requires the `0x`, then why do you believe we should trim the `0x`?

Reply:
Hardhat doesn't require the `0x` but assumes they are there. Hardhat later on trims the `0x` when normalising. Here is the relevant code:
https://github.com/NomicFoundation/hardhat/blob/024d72b09c6edefb00c012e9514a0948c255d0ab/v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/utils/convert-to-edr.ts#L192-L195

When running the `hardhart-tests` for example, I confirmed this expected format from the trace generated from Geth which the tests depend on: 
- https://github.com/NomicFoundation/edr/blob/main/hardhat-tests/test/fixture-debug-traces/traceCall/contractAGetMessage.ts
- https://github.com/NomicFoundation/edr/blob/main/hardhat-tests/test/fixture-debug-traces/traceCall/contractAGetMessageDebugConfig.ts